File: 215. Kth Largest Element in an Array.cpp
// 215. Kth Largest Element in an Array

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> hp;
        
        for (int n : nums) {
            hp.push(n);
            if (hp.size() > k) {
                hp.pop();
            }
        }
        return hp.top();
    }
};

File: 215. Kth Largest Element in an Array.py
# 215. Kth Largest Element in an Array


class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        hp = []
        for n in nums:
            heappush(hp, n)
            if len(hp) > k:
                heappop(hp)
        return heappop(hp)
